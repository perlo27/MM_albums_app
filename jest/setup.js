import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

global.fetch = require('jest-fetch-mock');

jest.mock('react-navigation-fluid-transitions', () => ({
  'Transition': 'Transition',
  'FluidNavigator': jest.fn()
}));

jest.mock('native-base', () => ({
  'Header': 'Header',
  'Body': 'Body',
  'Left': 'Left',
  'Button': 'Button',
  'Icon': 'Icon',
  'Spinner': 'Spinner'
}));