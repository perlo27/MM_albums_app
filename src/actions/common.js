export const REQUEST = 'REQUEST';
export const SUCCESS = 'SUCCESS';
export const ERROR = 'ERROR';

export const requestType = name => `${name}_${REQUEST}`;
export const successType = name => `${name}_${SUCCESS}`;
export const errorType = name => `${name}_${ERROR}`;

export const generateAction = name => ({
  request: () => ({ type: requestType(name) }),
  success: ({response}) => ({
    type: successType(name),
    response,
  }),
  failure: error => ({ type: errorType(name), error }),
});
