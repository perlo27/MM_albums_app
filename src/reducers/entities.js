import { GET_ALBUMS, GET_PHOTOS, successType, requestType } from '../actions';

const initialState = {
  albums: null,
  photos: null,
  areAlbumsLoading: false,
  arePhotosLoading: false,
};

export default (state = initialState, { type, response }) => {
  switch (type) {
    case requestType(GET_ALBUMS):
      return { ...state, areAlbumsLoading: true };
    case successType(GET_ALBUMS):
      return { ...state, albums: response, areAlbumsLoading: false };
    case successType(GET_PHOTOS):
      return { ...state, photos: response, arePhotosLoading: false,};
    default:
      return state;
  }
};
