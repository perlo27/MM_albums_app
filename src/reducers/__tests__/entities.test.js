import entities, {initialState} from '../entities';
import { getAlbumsAction, getPhotosAction } from '../../actions';
import { initState  } from '../../../jest/moks';

const testAlbumsRequestAction = getAlbumsAction.request();
const testAlbumsSuccessAction = getAlbumsAction.success({response: initState.entities.albums});
const testPhotosRequestAction = getAlbumsAction.request();
const testPhotosSuccessAction = getPhotosAction.success({response: initState.entities.photos});

describe('entities', () => {
  it('should handle getAlbumsAction request action properly', () => {
    expect(entities(initialState, testAlbumsRequestAction)).toMatchSnapshot();
  });
  it('should handle getAlbumsAction success action properly', () => {
    expect(entities(initialState, testAlbumsSuccessAction)).toMatchSnapshot();
  });
  it('should handle getPhotosAction request action properly', () => {
    expect(entities(initialState, testPhotosRequestAction)).toMatchSnapshot();
  });
  it('should handle getPhotosAction success action properly', () => {
    expect(entities(initialState, testPhotosSuccessAction)).toMatchSnapshot();
  });
});