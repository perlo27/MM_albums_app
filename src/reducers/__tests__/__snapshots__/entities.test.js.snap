// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entities should handle getAlbumsAction request action properly 1`] = `
Object {
  "albums": null,
  "areAlbumsLoading": true,
  "arePhotosLoading": false,
  "photos": null,
}
`;

exports[`entities should handle getAlbumsAction success action properly 1`] = `
Object {
  "albums": Array [
    Object {
      "id": 1,
      "title": "Quidem molestiae enim",
      "userId": 1,
    },
    Object {
      "id": 2,
      "title": "Quidem molestiae enim 1",
      "userId": 2,
    },
  ],
  "areAlbumsLoading": false,
  "arePhotosLoading": false,
  "photos": null,
}
`;

exports[`entities should handle getPhotosAction request action properly 1`] = `
Object {
  "albums": null,
  "areAlbumsLoading": true,
  "arePhotosLoading": false,
  "photos": null,
}
`;

exports[`entities should handle getPhotosAction success action properly 1`] = `
Object {
  "albums": null,
  "areAlbumsLoading": false,
  "arePhotosLoading": false,
  "photos": Object {
    "1": Array [
      Object {
        "albumId": 1,
        "id": 1,
        "thumbnailUrl": "thumbnailUrl",
        "title": "Test",
        "url": "url",
      },
    ],
    "2": Array [
      Object {
        "albumId": 2,
        "id": 2,
        "thumbnailUrl": "thumbnailUrl",
        "title": "Test2",
        "url": "url2",
      },
    ],
  },
}
`;
